### PR TITLE
SNOW-1333163: Add support for default proxy configuration

### PR DESCRIPTION
**Pull request by vc-74**
### Description
Although the current code supports HTTP proxies, it does not allow to use the [default proxy configuration](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultproxy?view=net-8.0).

This PR allows using the default proxy by specifying the `useProxy=true` parameter in the connection string without specifying an explicit proxy (`proxyHost`, `proxyPort`...).

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`) (I only executed `HttpUtilTest` to avoid having to configure a connection to Snowflake)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Comment by vc-74** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 1]
```diff
@@ -444,4 +444,4 @@ public void AfterTest(ITest test)
 
         public ActionTargets Targets => ActionTargets.Test | ActionTargets.Suite;
     }
-}
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/SFBaseTest.cs:447]: actually we leave 1 new line at the end of the file for purpose so I would prefer not to delete it

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/SFBaseTest.cs (hunk 1 above)
Comment (vc-74) [reply Snowflake.Data.Tests/SFBaseTest.cs]: Fixed

File: Snowflake.Data/Core/Session/SFSessionHttpClientProxyProperties.cs [hunk 2]
```diff
@@ -22,7 +23,13 @@ internal class Extractor : IExtractor
             public SFSessionHttpClientProxyProperties ExtractProperties(SFSessionProperties propertiesDictionary)
             {
                 var properties = new SFSessionHttpClientProxyProperties();
-                if (Boolean.Parse(propertiesDictionary[SFSessionProperty.USEPROXY]))
+
+                if (propertiesDictionary.TryGetValue(SFSessionProperty.USEPROXY, out string stringUseProxy))
```
[the remark this answers is not in the document]
Comment (vc-74) [reply Snowflake.Data/Core/Session/SFSessionHttpClientProxyProperties.cs]: Fixed

File: Snowflake.Data/Snowflake.Data.csproj [hunk 3]
```diff
@@ -13,11 +13,11 @@
     <Product>Snowflake Connector for .NET</Product>
     <Authors>howryu, tchen</Authors>
     <Copyright>Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.</Copyright>
-    <Version>3.1.0</Version>
+    <Version>3.2.0</Version>
```
[the remark this answers is not in the document]
Comment (vc-74) [reply Snowflake.Data/Snowflake.Data.csproj]: Fixed

**Comment by sfc-gh-knozderko** [comment]
Hello @vc-74. Please add here a comment with text: "I have read the CLA Document and I hereby sign the CLA".

I also created a new PR, because I needed to update it and add some new logic: https://github.com/snowflakedb/snowflake-connector-net/pull/997 

Please review the new PR and add also a comment in it: "I have read the CLA Document and I hereby sign the CLA".

**Pull request closed without merging**